The single failing matrix leg — `test-wheel-ubuntu (ubuntu-22.04, 3.12)` — died on a CI-runner disk-space issue, not on this change:

```
##[error]No space left on device : '/home/runner/actions-runner/cached/2.335.1/_diag/pages/...log'
```

It ran out of space during a `torch` reinstall in the test env. Evidence this is infra-only and unrelated to the diff:

- The same wheel test suite **passed** on `test-wheel-ubuntu (ubuntu-24.04, 3.10)`.
- **All** build legs are green: `build (3.10/3.12)`, `build-flags`, `build-musa`, `build-wheel-cu13`, the Docker image, and the Ascend build — so the change compiles across every platform/flag combo.
- The other two wheel legs show as `cancelled` (matrix fail-fast after the disk-space leg tripped), not as genuine failures.

The change itself is an 8-line, behavior-preserving `::tolower`/`::toupper` UB fix and can't cause disk exhaustion. Could a maintainer re-run the failed job (or apply `run-ci`) when convenient? Thanks!

--- a/mooncake-store/include/ha/oplog/oplog_store_factory.h
+++ b/mooncake-store/include/ha/oplog/oplog_store_factory.h
@@ -33,7 +33,7 @@ static constexpr OpLogStoreType kDefaultOpLogStoreType =
 inline OpLogStoreType ParseOpLogStoreType(const std::string& type_str) {
     std::string lower = type_str;
     std::transform(lower.begin(), lower.end(), lower.begin(),
-                   [](unsigned char c) { return std::tolower(c); });
+                   [](unsigned char c) -> char { return std::tolower(c); });
     if (lower == "localfs" || lower == "local_fs") {
         return OpLogStoreType::LOCAL_FS;
     }

--- a/mooncake-store/include/utils.h
+++ b/mooncake-store/include/utils.h
@@ -223,7 +223,7 @@ std::string expected_to_str(const tl::expected<T, ErrorCode>& expected) {
 
     // Convert to uppercase for comparison
     std::transform(unit.begin(), unit.end(), unit.begin(),
-                   [](unsigned char c) { return std::toupper(c); });
+                   [](unsigned char c) -> char { return std::toupper(c); });
 
     // Apply unit multiplier
     const double KB = 1024.0;

--- a/mooncake-transfer-engine/src/transport/tcp_transport/tcp_transport.cpp
+++ b/mooncake-transfer-engine/src/transport/tcp_transport/tcp_transport.cpp
@@ -605,7 +605,7 @@ TcpTransport::TcpTransport() : context_(nullptr), running_(false) {
     if (getenv("MC_TCP_ENABLE_CONNECTION_POOL") != nullptr) {
         std::string val(getenv("MC_TCP_ENABLE_CONNECTION_POOL"));
         std::transform(val.begin(), val.end(), val.begin(),
-                       [](unsigned char c) { return std::tolower(c); });
+                       [](unsigned char c) -> char { return std::tolower(c); });
         if (val == "0" || val == "false" || val == "no") {
             enable_connection_pool_ = false;
         } else {

